command/alloc_fs.go

```diff
@@ -398,6 +398,12 @@ func getRandomJobAlloc(client *api.Client, jobID string) (string, error) {
 	}
 
 	r := rand.New(rand.NewSource(time.Now().UnixNano()))
-	allocID := runningAllocs[r.Intn(len(runningAllocs))].ID
-	return allocID, err
+	alloc := runningAllocs[r.Intn(len(runningAllocs))]
+
+	return alloc, err
+}
+
+func getRandomJobAllocID(client *api.Client, jobID string) (string, error) {
+	alloc, err := getRandomJobAlloc(client, jobID)
+	return alloc.ID, err
```

Comment:
This looks like it will panic on an error from getRandomJobAlloc

Reply:
That's true! I'll update it.